tools/autograd/gen_python_functions.py

```diff
@@ -702,8 +702,8 @@ def group_overloads(declarations, is_python_method):
     for x, dictionary in sorted(grouped.items()):
         if 'base' not in dictionary:
             raise RuntimeError(
-                "'base' not in dictionary for {}. keys are {}".format(
-                    x, list(dictionary.keys())))
+                "_out variant function {} can not find matched non-out 'base' function "
+                "{} in dictionary.".format(dictionary['signature'], x))
```

Comment:
Let's try to make this more human readable:
```
While identifying overloads, we found an out= schema {out_schema} without a corresponding non-out variant.
We expected the non-out variant to have schema {expected_schema}. Please check that you spelled the schema correctly in native_functions.yaml.
```

I think the `declarations` argument to gen_python_functions might have possible candidates. If so, we can make the error message even more prescriptive:

```
While identifying overloads, we found an out= schema {out_schema} without a corresponding non-out variant.
We expected the non-out variant to have schema {expected_schema}. Please check that you spelled the schema correctly in native_functions.yaml. We discovered the following candidates:
- {candidate1}
- {candidate2}
...
```

Reply:
> I think the `declarations` argument to gen_python_functions might have possible candidates. If so, we can make the error message even more prescriptive:

Yes, it's doable. And the overhead for a whole loop through `declarations` to find the non-out function is minimum.